src/sql/parts/notebook/models/notebookModel.ts

```diff
@@ -394,6 +394,10 @@ export class NotebookModel extends Disposable implements INotebookModel {
 			}
 			let profile = new ConnectionProfile(this._notebookOptions.capabilitiesService, this.connectionProfile);
 
+			// TODO: this code needs to be fixed since it is called before the this._savedKernelInfo is set.
```

Comment:
This TODO should be handled by Yurong's in progress work. Rather than duplicating the fix I'll wait until hers is in and then re-test.